src/mono/mono/mini/interp/transform.c

```diff
@@ -6571,7 +6571,11 @@ generate_code (TransformData *td, MonoMethod *method, MonoMethodHeader *header,
 						MonoClass *field_class = mono_class_from_mono_type_internal (ftype);
 						interp_emit_metadata_update_ldflda (td, field, error);
 						goto_if_nok (error, exit);
```

Comment:
Looks like the whole code following here should be replaced with `interp_emit_ldobj`

Reply:
@BrzVlad thanks! I also added `BARRIER_IF_VOLATILE`